packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -30,27 +39,176 @@ export default class FulfillmentModuleService<
       FulfillmentSet: { dto: FulfillmentTypes.FulfillmentSetDTO }
       ServiceZone: { dto: FulfillmentTypes.ServiceZoneDTO }
       ShippingOption: { dto: FulfillmentTypes.ShippingOptionDTO }
+      GeoZone: { dto: FulfillmentTypes.GeoZoneDTO }
     }
   >(FulfillmentSet, generateMethodForModels, entityNameToLinkableKeysMap)
   implements IFulfillmentModuleService
 {
   protected baseRepository_: DAL.RepositoryService
-  protected readonly fulfillmentService_: ModulesSdkTypes.InternalModuleService<TEntity>
+  protected readonly fulfillmentSetService_: ModulesSdkTypes.InternalModuleService<TEntity>
+  protected readonly serviceZoneService_: ModulesSdkTypes.InternalModuleService<TServiceZoneEntity>
+  protected readonly geoZoneService_: ModulesSdkTypes.InternalModuleService<TGeoZoneEntity>
 
   constructor(
-    { baseRepository, fulfillmentService }: InjectedDependencies,
+    {
+      baseRepository,
+      fulfillmentSetService,
+      serviceZoneService,
+      geoZoneService,
+    }: InjectedDependencies,
     protected readonly moduleDeclaration: InternalModuleDeclaration
   ) {
     // @ts-ignore
     super(...arguments)
     this.baseRepository_ = baseRepository
-    this.fulfillmentService_ = fulfillmentService
+    this.fulfillmentSetService_ = fulfillmentSetService
+    this.serviceZoneService_ = serviceZoneService
+    this.geoZoneService_ = geoZoneService
   }
 
   __joinerConfig(): ModuleJoinerConfig {
     return joinerConfig
   }
 
+  /**
+   * Returns the identifier of a geo zone. The identifier is a string that is
+   * generated based on the type of the geo zone and the properties of the geo
+   * zone. The identifier is used for map building and retrieval.
+   *
+   * @param geoZone
+   * @param preventIdUsage
+   * @protected
+   */
+  protected static getGeoZoneIdentifier(
```

Comment:
suggestion: identifier can be an `[]` and each value has a string to identify the attribute. The method returns `identifier.join("_")`. 
```
if ("country_code" in geoZone && geoZone.country_code) {
  identifier.push("c:" + geoZone.country_code)
}
```
that will avoid clashing in case both properties has the same value. like `city: SP, province_code: SP`

Reply:
With that example and the current code, the identifier will be spsp, which shouldn't conflict no? But i can do that still 😊

Reply:
they are optional no?
if you have city: sp, and other one province_code: sp, they will become the same.

Reply:
I will add a validator to ensure the data, but the type is included in the identifier.
If one has city: sp, then it is probably of type city since there is no province code, and province code would have to be provided if it was of type province, but in that case city is not expected